components/search/BookingInfoSearch/BookingInfoSearchUIComplete.tsx

```diff
@@ -1,57 +1,97 @@
 "use client";
+import { useState } from "react";
 import AdultsNumberInput from "./SearchComponents/AdultsNumberInput";
 import { DatePickerWithRange } from "./DatePickerWithRange";
-
-type BookingInfo = {
-  checkinDate: string;
-  checkoutDate: string;
-  adultsNumber: number;
-  numDays: number;
-};
+import { searchParamsType } from "@/app/page";
 
 type BookingInfoUISearchCompleteProps = {
-  bookingInfo: BookingInfo;
-  setBookingInfo: (info: BookingInfo) => void;
+  bookingInfo: searchParamsType;
+  setBookingInfo: (info: searchParamsType) => void;
 };
 
-const BookingInfoUISearchComplete: React.FC<BookingInfoUISearchCompleteProps> = ({
-  bookingInfo,
-  setBookingInfo,
-}) => {
+const BookingInfoUISearchComplete: React.FC<
+  BookingInfoUISearchCompleteProps
+> = ({ bookingInfo, setBookingInfo }) => {
+  const [tempBookingInfo, setTempBookingInfo] = useState(bookingInfo);
+  const [isDateValid, setIsDateValid] = useState(true);
+  const [isValid, setIsValid] = useState(true);
+
   const handleDateChange = (dates: {
     checkinDate: string;
     checkoutDate: string;
     numDays: number;
   }) => {
-    setBookingInfo({ ...bookingInfo, ...dates });
+    setTempBookingInfo((prev) => ({ ...prev, ...dates }));
+  };
+
+  const handleAdultsNumberChange = (adults: number | null) => {
+    if (adults !== null) {
+      setIsValid(true);
+      setTempBookingInfo((prev) => ({ ...prev, adultsNumber: adults }));
+    } else {
+      setIsValid(false);
+    }
   };
 
-  const handleAdultsNumberChange = (adults: number) => {
-    setBookingInfo({ ...bookingInfo, adultsNumber: adults });
+  const handleApplyFilters = () => {
+    if (isValid) {
+      setBookingInfo(tempBookingInfo);
+    }
+  };
+
+  const handleRevertChanges = () => {
+    const confirmRevert = window.confirm(
+      "Are you sure you want to revert your booking changes?"
+    );
+    if (confirmRevert) {
+      setTempBookingInfo(bookingInfo);
+      setBookingInfo(tempBookingInfo);
```

Comment:
```suggestion
```

Reply:
At that point tempBookingInfo is already bookingInfo so this wouldn't change anything or trigger a rerender.